app/models/spree/user_decorator.rb

```diff
@@ -0,0 +1,10 @@
+Spree::User.class_eval do
+
+  delegate :firstname, :lastname, :address, :phone, to: :bill_address, allow_nil: true
+
+  scope :with_complete_orders, -> { joins(:orders).merge(Spree::Order.complete) }
+  scope :without_complete_orders, -> { where.not(id: with_complete_orders.pluck(:id)) }
```

Comment:
it seems to fire an extra query. first it will pluck with_complete_orders and then it will fire another query